### PR TITLE
pci-3852: policy checker handler: add logging

### PR DESCRIPTION
**Pull request by aliculPix4D**
policy checker handler: 
- minor improvements to the policy checker error messages
- adds logging

### CODE REVIEW

File: atc/api/policychecker/handler.go [hunk 1]
```diff
@@ -34,10 +34,14 @@ type policyCheckingHandler struct {
...
 	acc := accessor.GetAccessor(r)
 
+	logger := h.logger.Session("policy-check-handler")
 	result, err := h.policyChecker.Check(h.action, acc, r)
 	if err != nil {
 		w.WriteHeader(http.StatusBadRequest)
-		fmt.Fprintf(w, "policy check error: %s", err.Error())
+		// error should be logged and meaningful to the Concourse operator/maintainer
+		logger.Error("policy-checker", err)
+		// actual error is not useful and can be confussing to Concourse user
```
Comment (marco-m-pix4d) [inline atc/api/policychecker/handler.go:43]: 1 s

**Comment by aliculPix4D** [comment]
FYI: the upstream tests are really quite flaky...

https://builder.ci.pix4d.com/teams/developers/pipelines/concourse-fork-pci-3852-policy-checker-add-logging/jobs/upstream-unit/builds/1

File: atc/api/policychecker/handler.go (hunk 1 above)
Comment (marco-m-pix4d) [reply atc/api/policychecker/handler.go:43]: We are using 3 names for the same thing:
- "policy-check-handler"
- "policy-checker"
- "policy checker"

This is confusing and will make it harder to grep for it. Can we use 1 instead?

Also, to help looking for the error from the logs, we could do something similar to this:

```go
userMsg := "policy-checker: unreachable or misconfigured"
...
logger.Error("policy-checker", err, "user-msg": userMsg)
...
fmt.Fprintf(w, userMsg)
```

**Review by marco-m-pix4d — APPROVED** [review]

**Comment by marco-m-pix4d** [comment]
> always the same test suite and always the same data race...

You can remove the `-race` from the ginko invocation (in a separate PR).
The flake (or real bug) will stay there, but as long as we do not touch that part of the code, it is useless and not our "fault"...

File: atc/policy/opa/opa.go [hunk 2]
```diff
@@ -60,23 +60,23 @@ func (c opa) Check(input policy.PolicyCheckInput) (policy.PolicyCheckResult, err
 	client.Timeout = c.config.Timeout
 	resp, err := client.Do(req)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("OPA server: connecting: %w", err)
```
Comment (iAmoric) [inline atc/policy/opa/opa.go:63]: Is the wording correct?
For me when I read `connecting` I first think "it's trying to connect... the error will come later if any".
"connection error" or "error while connecting" ? 

(very minor, just an open question, I'm also fine keeping this as it is)

**Review by iAmoric — APPROVED** [review]

File: atc/api/policychecker/handler.go (hunk 1 above)
Comment (aliculPix4D) [reply atc/api/policychecker/handler.go]: Done as suggested...

File: atc/policy/opa/opa.go (hunk 2 above)
Comment (marco-m-pix4d) [reply atc/policy/opa/opa.go:63]: It is based on convention, what is important is uniformity in usage. In this case, the classic example is:
```go
fi, err := os.Open(...)
if err != nil {
    return fmt.Errorf("opening ...", ...)
}
```
In an error message, the gerund means "while I was doing X"
In a _debug_ log message, the gerund means: "I started doing X"

(while at it, note that Go does not want the word "error" appear in error messages).

Comment (aliculPix4D) [reply atc/policy/opa/opa.go:63]: valid point... I agree with the reasoning... Let me check from where I got the "connecting..."

Comment (aliculPix4D) [reply atc/policy/opa/opa.go:63]: OK. I think the comment from @marco-m-pix4d answers the dilemma. It seems we are keeping the message as it is.

Comment (iAmoric) [reply atc/policy/opa/opa.go:63]: Ok, thanks. Good to know :)

File: atc/api/policychecker/handler.go (hunk 1 above)
Comment (aliculPix4D) [reply atc/api/policychecker/handler.go]: Ah... now, I know why I had the `-handler` in the session name:

> {"timestamp":"2024-10-09T10:05:31.196402292Z","level":"error","source":"atc","message":"atc.policy-checker.policy-checker","data":{"error":"OPA server: connecting: Post \"http://localhost:8181/v1/data/concourse/decision\": dial tcp [::1]:8181: connect: connection refused","session":"134","user-msg":"policy-checker: unreachable or misconfigured"}}

note the msg: `atc.policy-checker.policy-checker`. The `atc.policy-checker` is already there. So the session name should just be `handler`

**Pull request merged**